src/core/qgselevationmap.cpp

```diff
@@ -0,0 +1,90 @@
+/***************************************************************************
+  qgselevationmap.cpp
+  --------------------------------------
+  Date                 : August 2022
+  Copyright            : (C) 2022 by Martin Dobias
+  Email                : wonder dot sk at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgselevationmap.h"
+
+#include "qgsrasterblock.h"
+
+#include <QPainter>
+#include <algorithm>
+#include <cmath>
+
+
+QgsElevationMap::QgsElevationMap( const QSize &size )
+  : mElevationImage( size, QImage::Format_ARGB32 )
+{
+  mElevationImage.fill( 0 );
+  mPainter.reset( new QPainter );
+  mPainter->begin( &mElevationImage );
```

Comment:
Won't destroying a painter in an active state cause issues? (via the default destructor)

Reply:
It seems that should be fine - if you are referring to a missing QPainter::end() call - it is optional according to the docs https://doc.qt.io/qt-5/qpainter.html#end

Reply:
Hmm that's odd! I often get crashes from python code when painters are destroyed while still in use.. maybe that's a PyQt "thing" though 🤨